### PR TITLE
Adding support for DS1621

### PR DESCRIPTION
**Pull request by DoNotTapeOver**
## Description
Adding support for DS1621 thermometer.

## Motivation and Context
nanoFramework,Iot.Device does not provide support for this chips yet.

## How Has This Been Tested?
The sample code was tested on a DS1621 with a ESP32-WROOM-32.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [x] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [x] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

**Comment by josesimoes** [comment]
@DoNotTapeOver shouldn't the title be Ds1621?

**Comment by DoNotTapeOver** [comment]
Ha, yes it should. Funny story, the '6' on my keyboard has stopped working, I need to remember that when I type now...

File: devices/Ds1621/Ds1621.nuspec [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    <projectUrl>https://github.com/nanoframework/nanoFramework.IoT.Device</projectUrl>
+    <icon>images\nf-logo.png</icon>
+    <repository type="git" url="https://github.com/nanoframework/nanoFramework.IoT.Device" commit="$commit$" />
+    <copyright>Copyright (c) .NET Foundation and Contributors</copyright>
+    <description>This package includes the .NET IoT Core binding Iot.Device.Ds1621 for .NET nanoFramework C# projects.</description>
+    <summary>Iot.Device.Ds1621 assembly for .NET nanoFramework C# projects</summary>
+    <tags>nanoFramework C# csharp netmf netnf Iot.Device.Ds1621</tags>
+    <dependencies>
+      <dependency id="nanoFramework.CoreLibrary" version="1.12.0" />
+      <dependency id="nanoFramework.Device.OneWire" version="1.3.4" />
```
Comment (networkfusion) [inline devices/Ds1621/Ds1621.nuspec:24]: Requires extra dependencies, most likely cause of the build error. Take them from the `.nfproj`.

Comment (josesimoes) [reply devices/Ds1621/Ds1621.nuspec:24]: If the build is failing on nuget validation, the tool is outputting exactly what is missing.

**Comment by networkfusion** [comment]
@DoNotTapeOver I think it also needs UnitsNet.nanoFramework.Temperature.4.140.0 in the nuspec.

File: devices/Ds1621/Ds1621.nuspec (hunk 1 above)
Comment (networkfusion) [reply devices/Ds1621/Ds1621.nuspec]: Internal chat noted. The issue does involve adding these to the nuspec.

**Comment by DoNotTapeOver** [comment]
Is there anyway to check the build for errors on my end without pushing an update?

**Comment by networkfusion** [comment]
> Is there anyway to check the build for errors on my end without pushing an update?

If the sln builds on your local machine, then it generally comes down to issues that makes the CI happy. In CI cases, you really need to check the CI and see why it is not happy (checks->View more details on Azure Pipelines)
for instance, your last build (https://dev.azure.com/nanoframework/IoT.Device/_build/results?buildId=35372&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=d1e88c09-dceb-53c1-5579-19c846f3526a)
suggests there are issues against the test framework.

File: devices/Ds1621/tests/NFUnitTest.nfproj [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  </ItemGroup>
+  <ItemGroup>
+    <None Include="nano.runsettings" />
+    <None Include="packages.config" />
+  </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\Ds1621.nfproj" />
+  </ItemGroup>
+  <Import Project="$(NanoFrameworkProjectSystemPath)NFProjectSystem.CSharp.targets" Condition="Exists('$(NanoFrameworkProjectSystemPath)NFProjectSystem.CSharp.targets')" />
+  <!-- MANUAL UPDATE HERE -->
```
Comment (networkfusion) [inline devices/Ds1621/tests/NFUnitTest.nfproj:77]: A manual change is required here to ensure you are targeting the correct "unit test nuget". For ease, you could downgrade the nuget to the expected version...

Comment (networkfusion) [reply devices/Ds1621/tests/NFUnitTest.nfproj:77]: You might have beat me to this 😅

Comment (DoNotTapeOver) [reply devices/Ds1621/tests/NFUnitTest.nfproj:77]: Only by a second or two. Thank you for they help though, I do appreciate it.

File: devices/Ds1621/tests/Ds1621Tests.cs [hunk 3]
```diff
@@ -0,0 +1,79 @@
+using Iot.Device.Common;
```
Comment (Ellerbach) [inline devices/Ds1621/tests/Ds1621Tests.cs:1]: please add the standard header here as well

File: devices/Ds1621/tests/TestHelper.cs [hunk 4]
```diff
@@ -0,0 +1,64 @@
+﻿using Iot.Device.Common;
```
Comment (Ellerbach) [inline devices/Ds1621/tests/TestHelper.cs:1]: please add the standard header here as well

**Review by Ellerbach — COMMENTED** [review]
Very minimal comments. For the rest, looks all good! We will be able to merge this one quite fast. Thanks a lot for the great work and taking into account other PR feedback. Makes the review fast and easy!

File: devices/Ds1621/packages.config [hunk 5]
```diff
@@ -0,0 +1,11 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="nanoFramework.CoreLibrary" version="1.12.0" targetFramework="netnano1.0" />
+  <package id="nanoFramework.Runtime.Events" version="1.10.0" targetFramework="netnano1.0" />
+  <package id="nanoFramework.System.Device.Gpio" version="1.0.4" targetFramework="netnano1.0" />
+  <package id="nanoFramework.System.Device.I2c" version="1.0.3" targetFramework="netnano1.0" />
+  <package id="nanoFramework.System.Math" version="1.4.4" targetFramework="netnano1.0" />
+  <package id="Nerdbank.GitVersioning" version="3.5.108" targetFramework="netnano1.0" developmentDependency="true" />
+  <package id="StyleCop.MSBuild" version="6.2.0" targetFramework="netnano1.0" developmentDependency="true" />
+  <package id="UnitsNet.nanoFramework.Temperature" version="4.140.0" targetFramework="netnano1.0" />
```
Comment (networkfusion) [inline devices/Ds1621/packages.config:10]: Update to `.144`

File: devices/Ds1621/samples/packages.config [hunk 6]
```diff
@@ -0,0 +1,10 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="nanoFramework.CoreLibrary" version="1.12.0" targetFramework="netnano1.0" />
+  <package id="nanoFramework.Hardware.Esp32" version="1.3.6.7" targetFramework="netnano1.0" />
+  <package id="nanoFramework.Runtime.Events" version="1.10.0" targetFramework="netnano1.0" />
+  <package id="nanoFramework.Runtime.Native" version="1.5.4" targetFramework="netnano1.0" />
+  <package id="nanoFramework.System.Device.Gpio" version="1.0.4" targetFramework="netnano1.0" />
+  <package id="nanoFramework.System.Device.I2c" version="1.0.3" targetFramework="netnano1.0" />
+  <package id="UnitsNet.nanoFramework.Temperature" version="4.140.0" targetFramework="netnano1.0" />
```
Comment (networkfusion) [inline devices/Ds1621/samples/packages.config:9]: Same...

**Review by Ellerbach — APPROVED** [review]
Thanks, looks good to me! Nice addition!

**Comment by Ellerbach** [comment]
@DoNotTapeOver the build is failing, please check what's the issue, once fixed, we will be able to merge. The branch has been updated with the latest changes from develop branch, so maybe something related to this (one of the nuget updated or something equivalent).

**Comment by josesimoes** [comment]
@DoNotTapeOver that build failure is on me. That's because of a recent change I've made in the pipeline template. I'll fix it asap.

**Pull request merged**